plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -301,8 +302,10 @@ private static HiveColumnHandle toPartitionValuesParsedField(HiveColumnHandle ad
                 addColumn.getComment());
     }
 
-    private DeltaLakeTransactionLogEntry buildCommitInfoEntry(ConnectorSession session, Block block, int pagePosition)
+    private DeltaLakeTransactionLogEntry buildCommitInfoEntry(ConnectorSession session, int pagePosition, Block... blocks)
```

Comment:
I don't think any of these methods need variable arguments as they're not explicitly implementing the CheckPointFieldExtractor interface. You could have these methods take exactly the needed number of Blocks as argument and let `createCheckPointFieldExtractor` adapt them as necessary. e.g. `case TRANSACTION -> (session, pagePosition, blocks) -> buildTxnEntry(session, pagePosition, blocks[0]);`